sdk/keyvault/keyvault-keys/src/lro/delete/operation.ts

```diff
@@ -0,0 +1,101 @@
+import { AbortSignalLike } from "@azure/abort-controller";
+import { PollOperationState, PollOperation } from "@azure/core-lro";
+import { RequestOptionsBase } from "@azure/core-http";
+import { DeletedKey, KeyClientInterface } from "../../keysModels";
+
+/**
+ * @interface
+ * An interface representing the state of a delete key's poll operation
+ */
+export interface DeleteKeyPollOperationState extends PollOperationState<DeletedKey> {
+  name: string;
+  requestOptions?: RequestOptionsBase;
+  client: KeyClientInterface;
+}
+
+/**
+ * @interface
+ * An interface representing a delete key's poll operation
+ */
+export interface DeleteKeyPollOperation
+  extends PollOperation<DeleteKeyPollOperationState, DeletedKey> {}
+
+/**
+ * @summary Reaches to the service and updates the delete key's poll operation.
+ * @param [options] The optional parameters, which are an abortSignal from @azure/abort-controller and a function that triggers the poller's onProgress function.
+ */
+async function update(
+  this: DeleteKeyPollOperation,
+  options: {
+    abortSignal?: AbortSignalLike;
+    fireProgress?: (state: DeleteKeyPollOperationState) => void;
+  } = {}
+): Promise<DeleteKeyPollOperation> {
+  const state = this.state;
+  const { name, client } = state;
+
+  const requestOptions = state.requestOptions || {};
+  if (options.abortSignal) {
+    requestOptions.abortSignal = options.abortSignal;
+  }
+
+  if (!state.started) {
+    const deletedKey = await client.deleteKey(name, requestOptions);
+    state.started = true;
+    if (!deletedKey.properties.recoveryId) {
+      state.completed = true;
+      state.result = deletedKey;
+    }
+  }
+
+  if (!state.completed) {
+    try {
+      state.result = await client.getDeletedKey(name, { requestOptions });
+      state.completed = true;
+    } catch (error) {
+      if (error.statusCode !== 404) {
```

Comment:
If a 403 would throw an error, you might not want to treat that as an error below (state.error). But as long as that doesn't throw downstream, perhaps it's not such a bad idea, i.e. success but with warning.